build-system/git.js

```diff
@@ -28,17 +27,21 @@ const {
 } = require('./travis');
 const {getStdout} = require('./exec');
 
-const commitLogMaxCount = 100;
-
 /**
- * Returns the merge base of the current branch off of master, regardless of
- * the running environment.
+ * Returns the commit at which the current branch was forked off of master.
+ * On Travis, there is an additional merge commit, so we must pick the first of
+ * the boundary commits (prefixed with a -) returned by git rev-list.
+ * On local branches, this is merge base of the current branch off of master.
  * @return {string}
  */
-exports.gitMergeBaseMaster = function() {
+exports.gitBranchCreationPoint = function() {
   if (isTravisBuild()) {
     const traviPrSha = travisPullRequestSha();
-    return getStdout(`git merge-base master ${traviPrSha}`).trim();
+    const boundaryCommits = getStdout(
+      `git rev-list --boundary ${traviPrSha}...master | grep "^-"`
```

Comment:
How did you come up with this approach?

Reply:
1. I read the docs at https://git-scm.com/docs/git-merge-base
2. The "see also" section led me to https://git-scm.com/docs/git-rev-list 
3. This led me to discover https://stackoverflow.com/questions/1527234/finding-a-branch-point-with-git

Reply:
👍. Nit, you can `| head -n 1` to get only the first match.

Reply:
Nice. Done in 778730acd098df69aff9370b6e4766901a808dd9.